src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -29,6 +30,26 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
     /// </summary>
     internal class CompletionHandler : IRequestHandler<LSP.CompletionParams, LSP.CompletionList?>
     {
+        /// <summary>
+        /// This sets the max list size we will return in response to a completion request.
+        /// If there are more than this many items, we will set the isIncomplete flag on the returned completion list.
+        /// </summary>
+        private const int MaxCompletionListSize = 1000;
```

Comment:
consider an option here.  esp. so we can tweak this in teh regisry if needed.

Reply:
@CyrusNajmabadi  Would a hidden registry setting be good enough?  I don't know that we would want to necessarily expose this in tools options

Reply:
Yup.  I didn't mean user facing.  just using our options framework.  that way if someone encounters a blocking issue we can give a way to tweak this.